### PR TITLE
Clean up init

### PR DESCRIPTION
**Pull request by hugelgupf**
- split up pkg/uroot/util into pkg/upath, pkg/mount, and pkg/libinit (you cannot import a package named "init", because it conflicts with init functions)
- remove defaultsh special case from init
- remove weirdo profile.d stuff -- why was that even there? bash support?

### CODE REVIEW

File: cmds/core/init/init.go [hunk 1]
```diff
@@ -13,18 +13,14 @@ package main
...
 	"log"
 	"os"
 	"os/exec"
-	"path"
-	"path/filepath"
-	"runtime"
-	"strings"
 	"syscall"
 
+	"github.com/u-root/u-root/pkg/libinit"
```
Comment (rminnich) [inline cmds/core/init/init.go]: is pkg/libinit kind of redundant, pkg being a kind of standin for "lib" anyway? is pkg/init ok?

File: cmds/core/init/init.go [hunk 2]
```diff
@@ -75,62 +74,14 @@ func main() {
...
+	libinit.SetEnv()
 	// Create the root file systems.
-	util.Rootfs()
+	libinit.CreateRootfs()
+	libinit.NetInit()
 
 	envs = os.Environ()
 	debug("envs %v", envs)
 
-	// install /env.
```
Comment (rminnich) [inline cmds/core/init/init.go]: where did some of this stuff go? It seems to be gone and some of it was needed.

File: cmds/core/init/init.go (hunk 1 above)
Comment (hugelgupf) [reply cmds/core/init/init.go]: no, because `init` is a reserved word for init functions

File: cmds/core/init/init.go (hunk 2 above)
Comment (hugelgupf) [reply cmds/core/init/init.go]: what is it for? can you explain? it includes calls to `sudo` (we don't even have a sudo) and stuff that doesn't tie into anything here at all

Comment (rminnich) [reply cmds/core/init/init.go]: ah ha, now it all comes clear with your comment! it was for when we integrated u-root into other initramfs like tinycore ... it's very good that this is gone :-)

File: files/etc/profile.d/uroot.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+export PATH="/go/bin:/ubin:$PATH/buildbin:/bbin"
+
+# The IFS lets us force a rehash every time we type a command, so that when we
```
Comment (rminnich) [inline files/etc/profile.d/uroot.sh]: we probably need a README in this directory maybe? 
README
The uroot.sh can be included with -files when we integrate a u-root image into an exiting initramfs such as tinycore.

**Pull request merged**